domain/src/main/java/mihon/domain/chapter/interactor/GetChaptersToDownload.kt

```diff
@@ -0,0 +1,90 @@
+package mihon.domain.chapter.interactor
+
+import android.database.sqlite.SQLiteException
+import logcat.LogPriority
+import tachiyomi.core.common.util.system.logcat
+import tachiyomi.domain.category.interactor.GetCategories
+import tachiyomi.domain.chapter.model.Chapter
+import tachiyomi.domain.chapter.repository.ChapterRepository
+import tachiyomi.domain.download.service.DownloadPreferences
+import tachiyomi.domain.manga.model.Manga
+
+/**
+ * Interactor responsible for determining which chapters of a manga should be downloaded.
+ *
+ * @property chapterRepository The repository for accessing chapter data.
+ * @property downloadPreferences User preferences related to chapter downloads.
+ * @property getCategories Interactor for retrieving categories associated with a manga.
+ */
+class GetChaptersToDownload(
+    private val chapterRepository: ChapterRepository,
+    private val downloadPreferences: DownloadPreferences,
+    private val getCategories: GetCategories,
+) {
+
+    /**
+     * Determines chapters that should be downloaded based on user preferences.
+     *
+     * @param manga The manga for which chapters may be downloaded.
+     * @param newChapters The list of new chapters available for the manga.
+     * @return A list of chapters that should be downloaded.
+     */
+    suspend fun await(manga: Manga, newChapters: List<Chapter>): List<Chapter> {
+        if (!shouldDownloadNewChapters(manga)) {
+            return emptyList()
+        }
+
+        val downloadUnreadChapters = downloadPreferences.downloadUnreadChaptersOnly().get()
+        return if (downloadUnreadChapters) {
+            newChapters.filter { isUnreadChapter(manga.id, it.chapterNumber) }
+        } else {
+            newChapters
+        }
```

Comment:
Instead of doing this just get the whole db chapters and compare against that. That's what I meant by keep using the logic inside `SyncChaptersWithSource` in previous review